config/opal_config_hwloc.m4

```diff
@@ -90,7 +90,14 @@ AC_DEFUN([OPAL_CONFIG_HWLOC], [
 
     AS_IF([test $pkg_config_happy -ne 0],
           [opal_hwloc_WRAPPER_LDFLAGS="$pkg_config_ldflags"
-           opal_hwloc_WRAPPER_LIBS="$pkg_config_libs"],
+           opal_hwloc_WRAPPER_LIBS="$pkg_config_libs"
+           AS_IF([test "$opal_hwloc_mode" != "internal"], [
+             OPAL_FLAGS_APPEND_UNIQ([opal_hwloc_LIBS], [$pkg_config_libs])
```

Comment:
This is not what we want to be doing.  Our pkg-config support is really not properly setup at this point, and we need to fix that rather than continuing to paste over the bad.  We're adding things into libs without testing and without using OPAL_CHECK_PACKAGE, which is where we're supposed to be setting libs/ldflags.  I'm not sure why this patch does anything to help the build; OPAL_CHECK_PACKAGE should have failed long before here.

Let's start with an actual design doc on how we want package configuration to work (especially with pkg-config present) and implement that, rather than trying different hacks.

Reply:
The case this is fixing is `--with-foo=external` rather than `--with-foo=/path/to/foo`. I'm pretty sure the later is not seeing this problem.

What happens is OPAL_CHECK_PACKAGE is finding the external headers/ect, but it isn't appending the -L path to the libs to LD_FLAGS. I can try to patch that path if you think it is a better way to go.

I'll take this commit out to get the other two in for now.